drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -8900,6 +8908,46 @@ void RenderingDeviceVulkan::_begin_frame() {
 	frames[frame].index = Engine::get_singleton()->get_frames_drawn();
 }
 
+VkSampleCountFlagBits RenderingDeviceVulkan::_ensure_supported_sample_count(TextureSamples p_requested_sample_count) const {
+	VkSampleCountFlags sample_count_flags = limits.framebufferColorSampleCounts & limits.framebufferDepthSampleCounts;
+	VkSampleCountFlagBits max_sample_count = VK_SAMPLE_COUNT_1_BIT;
+
+	if (sample_count_flags & VK_SAMPLE_COUNT_2_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_2_BIT;
+	}
+	if (sample_count_flags & VK_SAMPLE_COUNT_4_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_4_BIT;
+	}
+	if (sample_count_flags & VK_SAMPLE_COUNT_8_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_8_BIT;
+	}
+	if (sample_count_flags & VK_SAMPLE_COUNT_16_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_16_BIT;
+	}
+	if (sample_count_flags & VK_SAMPLE_COUNT_32_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_32_BIT;
+	}
+	if (sample_count_flags & VK_SAMPLE_COUNT_64_BIT) {
+		max_sample_count = VK_SAMPLE_COUNT_64_BIT;
+	}
+
+	return MIN(max_sample_count, rasterization_sample_count[p_requested_sample_count]);
```

Comment:
This should all be deleted now as it is right now the new code will never be reached

Reply:
Oops, I don't know how I overlooked that. Fixed :)